backend-api/src/functions/jwks/jwksBuilder/jwksBuilder.ts

```diff
@@ -108,37 +137,52 @@ export class JwksBuilder implements IJwksBuilder {
       });
     }
 
-    let publicKeyAsJwk: JsonWebKey;
+    return successResult(undefined);
+  }
+
+  private convertToJwk(publicKey: Uint8Array): Result<JsonWebKey> {
     try {
-      publicKeyAsJwk = createPublicKey({
-        key: Buffer.from(getPublicKeyOutput.PublicKey),
+      const publicKeyAsJwk = createPublicKey({
+        key: Buffer.from(publicKey),
         type: "spki",
         format: "der",
       }).export({ format: "jwk" });
+
+      return successResult(publicKeyAsJwk);
     } catch {
       return errorResult({
         errorMessage: "Error formatting public key as JWK",
         errorCategory: ErrorCategory.SERVER_ERROR,
       });
     }
+  }
 
-    if (keyUsage === "ENCRYPT_DECRYPT") {
-      const encryptionJwk: EncryptionJwk = {
-        ...publicKeyAsJwk,
-        use: "enc",
-        alg: "RSA-OAEP-256",
-        kid: keyId,
-      };
-      return successResult(encryptionJwk);
-    } else {
-      const signingJwk: SigningJwk = {
-        ...publicKeyAsJwk,
-        use: "sig",
-        alg: "ES256",
-        kid: keyId,
-      };
-      return successResult(signingJwk);
-    }
+  private createEncryptionJwk(
+    publicKeyAsJwk: JsonWebKey,
+    keyId: string,
+  ): Result<EncryptionJwk> {
```

Comment:
Nitpick: these two functions don't need to return results, as they can't fail

```suggestion
  ): EncryptionJwk {
```

Reply:
Removed result form function and wrapped calls instead